src/components/app/routes/loaders/updateActiveEnterpriseCustomerUserLoader.js

```diff
@@ -0,0 +1,69 @@
+import { redirect, generatePath } from 'react-router-dom';
+
+import ensureAuthenticatedUser from './ensureAuthenticatedUser';
+import { getEnterpriseAppData, makeEnterpriseLearnerQuery, updateUserActiveEnterprise } from './rootLoader';
+
+/**
+ * TODO
+ * @param {*} queryClient
+ * @returns
+ */
+export default function makeUpdateActiveEnterpriseCustomerUserLoader(queryClient) {
+  return async function updateActiveEnterpriseCustomerUserLoader({ params = {}, request }) {
+    const requestUrl = new URL(request.url);
+    const authenticatedUser = await ensureAuthenticatedUser(requestUrl);
+    const { username, userId, email: userEmail } = authenticatedUser;
+    const { enterpriseSlug: nextEnterpriseSlug } = params;
+
+    const linkedEnterpriseCustomersQuery = makeEnterpriseLearnerQuery(username, nextEnterpriseSlug);
+    const enterpriseLearnerData = await queryClient.fetchQuery(linkedEnterpriseCustomersQuery);
+    const {
+      activeEnterpriseCustomer,
+      allLinkedEnterpriseCustomerUsers,
+    } = enterpriseLearnerData;
+
+    // User has no active, linked enterprise customer; return early.
+    if (!activeEnterpriseCustomer) {
+      return null;
+    }
+
+    if (nextEnterpriseSlug !== activeEnterpriseCustomer.slug) {
+      // Otherwise, try to find the enterprise customer for the given slug and update it as the active
+      // enterprise customer for the learner.
+      const foundEnterpriseCustomerForSlug = allLinkedEnterpriseCustomerUsers.find(
+        enterpriseCustomerUser => enterpriseCustomerUser.enterpriseCustomer.slug === nextEnterpriseSlug,
+      );
+      if (foundEnterpriseCustomerForSlug) {
+        await updateUserActiveEnterprise({
+          enterpriseCustomer: foundEnterpriseCustomerForSlug.enterpriseCustomer,
+        });
+        queryClient.setQueryData(linkedEnterpriseCustomersQuery.queryKey, {
+          ...enterpriseLearnerData,
+          activeEnterpriseCustomer: foundEnterpriseCustomerForSlug.enterpriseCustomer,
+          allLinkedEnterpriseCustomerUsers: allLinkedEnterpriseCustomerUsers.map(
+            ecu => ({
```

Comment:
[nit/suggestion] I'd spell this out despite the lack of brevity it may cause

Reply:
Ah, I had shortened it to `ecu` (not ideal, I agree) due to ESLint max line warnings on line 47 below. I'm leaning towards leaving it as if vs. finding another variable name or further hacking around ESLint formatting.